recipes/ncbi-cxx-toolkit-public/26.0/conanfile.py

```diff
@@ -0,0 +1,1532 @@
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+#----------------------------------------------------------------------------
+    def requirements(self):
+        if not self.version in self.conan_data["sources"].keys():
+            raise ConanException("Invalid Toolkit version requested. Available: " + ' '.join(self.conan_data["sources"].keys()))
+        self.output.info("Analyzing requirements. Please wait...")
+        curdir = os.getcwd()
+        self._get_Source()
+        os.chdir(self.tk_tmp_tree)
+        shared = "ON" if self.options.shared else "OFF"
+        subprocess.run(["cmake", self.tk_src_tree,
+            "-DNCBI_PTBCFG_COLLECT_REQUIRES=ON",
+            "-DNCBI_PTBCFG_COLLECT_REQUIRES_FILE=req",
+            "-DBUILD_SHARED_LIBS='%s'" % shared,
+            "-DNCBI_PTBCFG_PROJECT_TARGETS='%s'" % self.options.targets,
+            "-DNCBI_PTBCFG_PROJECT_LIST='%s'" % self.options.projects],
+            stdout = subprocess.DEVNULL, stderr = subprocess.DEVNULL)
+        reqfile = os.path.join(self.tk_src_tree, "..", "req")
+        if os.path.isfile(reqfile):
+            self.all_NCBI_requires = set(open(reqfile).read().split())
+            for req in self.all_NCBI_requires:
+                pkg = self._translate_ReqKey(req)
+                if pkg is not None:
+                    print("Package requires ", pkg)
+                    self.requires(pkg)
+                    pkg = pkg[:pkg.find("/")]
+#ATTENTION
+                    if req == "MySQL":
+                        self.requires("openssl/1.1.1l@#98215f1d057178dfd8c867950e16a0fe")
+                    if req == "CASSANDRA":
+                        self.requires("openssl/1.1.1l@#98215f1d057178dfd8c867950e16a0fe")
+                    if req == "TIFF":
+                        self.requires("zstd/1.5.0@#2797bc9a304b2c45e2239c0d4ad15207")
+            os.remove(reqfile)
+        os.chdir(curdir)
+
+#----------------------------------------------------------------------------
+    def source(self):
+        if self.tk_tmp_tree == "":
+            self._get_Source()
+        shutil.move(os.path.join(self.tk_tmp_tree, "include"), ".")
+        shutil.move(os.path.join(self.tk_tmp_tree, self._src), ".")
+        if os.path.isdir(os.path.join(self.tk_tmp_tree, "scripts")):
+            shutil.move(os.path.join(self.tk_tmp_tree, "scripts"), ".")
+        if os.path.isdir(os.path.join(self.tk_tmp_tree, "doc")):
+            shutil.move(os.path.join(self.tk_tmp_tree, "doc"), ".")
+        self._remove_tmp_tree(self.tk_tmp_tree)
+
+#----------------------------------------------------------------------------
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.configure(source_folder=self._src)
+        if self.settings.os == "Windows":
+            self.run('cmake --build . %s -j 1' % cmake.build_config)
```

Comment:
why doesn't cmake build work here?

https://github.com/conan-io/conan-center-index/blob/aa023f3fd3247efb2b29fcb50209d523ac661ecd/recipes/diligent-core/all/conanfile.py#L138

Reply:
Occasionally Visual Studio compiler "runs out of heap space". This is a workaround.